ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+from typing import Optional
+from abc import ABC
+
+
+class RBACValidator(ABC):
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: dict = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        self._payload = payload if payload else dict()
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
```

Comment:
please define the action mapping as a static property on the class. Use the full dictionary when mappping the action, and use the `keys()` when adding RBACValidator to the list of validators (right now it is an identical usage, but only with the requst names, rewritten in plain)